akka-actor-typed/src/main/scala/akka/actor/typed/Behavior.scala

```diff
@@ -157,22 +156,45 @@ object Behavior {
       case _ => behavior
     }
 
+  @tailrec
+  private def startTail(
+      behavior: Behavior[Any],
+      stack: Seq[InterceptorImpl[Any, Any]],
+      ctx: TypedActorContext[Any],
+      isReducing: Boolean): Behavior[Any] = {
+    if (isReducing) {
+      if (stack.nonEmpty) {
+        val wrapped = stack.head
+        if (wrapped.nestedBehavior ne behavior)
+          startTail(wrapped.replaceNested(behavior), stack.tail, ctx, isReducing = true)
+        else
+          startTail(wrapped, stack.tail, ctx, isReducing = true)
+      } else {
+        behavior
+      }
+    } else {
+      behavior match {
+        case innerDeferred: DeferredBehavior[Any] @unchecked =>
+          startTail(innerDeferred(ctx), stack, ctx, isReducing = false)
+        case wrapped: InterceptorImpl[Any, Any] @unchecked =>
+          startTail(wrapped.nestedBehavior, wrapped +: stack, ctx, isReducing = false)
+        case _ =>
+          if (stack.nonEmpty) {
+            startTail(behavior, stack, ctx, isReducing = true)
+          } else {
+            behavior
+          }
+      }
+    }
+  }
```

Comment:
I don't see the point of changing this since it's replacing the jvm stack with collection stack, and the code is just more complicated

Reply:
Because of the existence of `InterceptorImpl`, using a stack is not avoidable. 
Yes, it's replacing the jvm thread's stack with a user defined heap stack, which is more efficient for the job and may not trigger the `StackOverflow` exception.